wallet/src/account/mod.rs

```diff
@@ -868,5 +939,66 @@ fn group_utxos_for_input<T, Grouped: Clone>(
     Ok(tokens_grouped)
 }
 
+pub fn tx_size_with_outputs(outputs: &[TxOutput]) -> usize {
+    let tx = SignedTransaction::new(
+        Transaction::new(1, vec![], outputs.into()).expect("should not fail"),
+        vec![],
+    )
+    .expect("should not fail");
+    serialization::Encode::encoded_size(&tx)
+}
+
+fn input_signature_size(destination: &Destination) -> usize {
+    let sig_hex = "003c002dd5ea8f05240394eb109b8e9b52716db2720cebbc5dd66394afdf761d2ef76dc8a292d3a44e4d9f1f8fd8e17dc404e317082f5a2ce8adafde01e766aaa6";
+    let sig_bin: Vec<u8> = hex::FromHex::from_hex(sig_hex).expect("valid hex");
+    let signature = Encoded::from_bytes_unchecked(sig_bin.as_slice()).decode();
+
+    let raw_signature = match destination {
+        Destination::Address(_) => AuthorizedPublicKeyHashSpend::new(
+            PrivateKey::new_from_entropy(crypto::key::KeyKind::Secp256k1Schnorr).1,
+            signature,
+        )
+        .encode(),
+        Destination::PublicKey(_) => AuthorizedPublicKeySpend::new(signature).encode(),
+        Destination::AnyoneCanSpend => {
+            let input_sig = InputWitness::NoSignature(None);
+            return serialization::Encode::encoded_size(&input_sig);
+        }
+        Destination::ScriptHash(_) | Destination::ClassicMultisig(_) => unimplemented!(),
+    };
+
+    let input_sig = InputWitness::Standard(StandardInputSignature::new(
+        SigHashType::default(),
+        raw_signature,
+    ));
+    serialization::Encode::encoded_size(&input_sig)
+}
+
+fn coin_and_token_output_fees(feerate: mempool::FeeRate) -> WalletResult<(Amount, Amount)> {
```

Comment:
I'm wondering... the FeeRate type is public, why are we not using that in RPC, and are using u128 there instead?

Reply:
fixed